src/pages/Homepage.jsx

```diff
@@ -1,70 +1,83 @@
-import { useState } from 'react';
+import { useCallback, useState } from 'react';
+import { v4 as uuid } from 'uuid';
 import TimerComponent from '../components/TimerComponent';
 import ActiveTeamsComponent from '../components/ActiveTeamsComponent';
 import EliminatedTeamsComponent from '../components/EliminatedTeamComponent';
+import findTeam from '../utils/index.js';
 
 const Homepage = () => {
   const [teams, setTeams] = useState([]);
   const [draggedIndex, setDraggedIndex] = useState(null);
 
   const handleAddTeam = (type) => {
-    setTeams([
-      ...teams,
+    setTeams((prevState) => [
+      ...prevState,
       {
+        id: uuid(),
```

Comment:
Most of the time you're using the index for identifying a team. This is hard for useing proper key-attributes (see comment above). Plus if we would decide to use a database for storing the teams due to scaling the project, it would be hard to refactor everything. So plan ahead and always use IDs.

Reply:
Noted. ID always, no matter how small the code is.